chapters/packages.tex

```diff
@@ -321,3 +321,132 @@ \section{External Resources}\label{external-resources}
+This file is a copy of the associated template file, but extended with the translations in the specified language.
+The pattern \filename{$\mathit{language}$} stands for the ISO 639-1 language code, e.g., \filename{de} or \filename{en}.
+\end{enumerate}
+
+The detailed format of these files is described in \href{https://www.gnu.org/software/gettext/manual/gettext.pdf}{GNU gettext}.
+For Modelica translations, only the keywords \lstinline!msgctxt!, \lstinline!msgid! and \lstinline!msgstr! are used, meaning that a translation entry looks like this:
+\begin{lstlisting}
+#: $\mathit{filename}$:$\mathit{lineNumber}$
+msgctxt $\mathit{messageContext}$
+msgid $\mathit{messageIdentifier}$
+msgstr $\mathit{messageTranslation}$
+\end{lstlisting}
+
+\begin{nonnormative}
+In the remainder of this section, several facts about the gettext specification are interleaved non-normatively for easy access to some of the gettext basics.
+Always refer to the external gettext specification for full detail or in case of doubt.
+
+All text strings are in double quotes and encoded with UTF-8 characters.
+Comments start with an \lstinline!#! and are continued until the end of line.
+Spaces outside strings are ignored and used as separators.
+
+The files consist of a header and a body.
+The header is marked with an empty \lstinline!msgid! and looks like this:
+\begin{lstlisting}
+# SOME DESCRIPTIVE TITLE.
+# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
+# This file is distributed under the same license as the PACKAGE package.
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
+#
+#, fuzzy
+msgid ""
+msgstr ""
+"Project-Id-Version: PACKAGE VERSION\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2019-03-15 10:52+0100\n"
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
+"Language: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+\end{lstlisting}
+All general terms in the header should be replaced by specific information.
+
+Following the header, there is one translation entry for each string to be translated.
+It can start with an optional comment describing the location (file name and line number) of the text to translate.
+Multiple occurences of the same string can be listed here, separated by space.
+\end{nonnormative}
+
+The $\mathit{messageContext}$ following the keyword \lstinline!msgctxt! shall be the full name of the Modelica class (e.g., \lstinline!"Modelica.Blocks.Math.Sin"!) where the text appears.
+Short class definitions do not appear here.
+Texts in such classes belong to the enclosing full class definition.
+
+The text string which shall be translated is used as $\mathit{messageIdentifier}$ (following the \lstinline!msgid! keyword), and shall contain the original string from the Modelica code.
+% Describe down-quoting and concatenation of Modelica string literals.
```

Comment:
This is not clear, and I'm not sure _down_-quoting helps.

Reply:
Basically down- and up-quoting do not seem to be standard terminology, and we should be clear about the order when combining concatenation with quoting.

Reply:
I agree that it would be nice if we could avoid talking about down- and up-quoting, and to me this comes so natural that it shouldn't need explanation. It's just that I have learned both from our discussions and elsewhere that this topic is a common source of confusion…

I don't see how the order would be unclear.

Reply:
Discussion to be continued under the [new suggestion by @HansOlsson](https://github.com/modelica/ModelicaSpecification/pull/2956#discussion_r871111571). Resolving.